src/python_testing/TC_TIMESYNC_2_2.py

```diff
@@ -72,6 +73,12 @@ async def test_TC_TIMESYNC_2_2(self):
             tolerance = timedelta(minutes=1)
         compare_time(received=utc_dut, utc=th_utc, tolerance=tolerance)
 
+        self.print_step(5, "Read time source")
+        if self.check_pics("TIMESYNC.S.A0002"):
+            if utc_dut_initial is NullValue:
+                source = await self.read_ts_attribute_expect_success(endpoint=endpoint, attribute=attributes.TimeSource)
```

Comment:
```suggestion
            source = await self.read_ts_attribute_expect_success(endpoint=endpoint, attribute=attributes.TimeSource)
            if utc_dut_initial is NullValue:
```
if I read the test plan right....

Reply:
I guess it depends on how you apply the column ordering in the test plan and whether the test step needs to be applied if the expected outcome won't use it. It doesn't much matter either way because this attribute is checked in the 2.1 test. English is so imprecise compare to code, we should just write tests in code and be done with it.

Reply:
> and whether the test step needs to be applied if the expected outcome won't use it.

Well, presumably if the read fails the test fails...